packages/lit-dev-tools-esm/src/build-unversioned-docs.ts

```diff
@@ -0,0 +1,154 @@
+ */
+const buildAndTransformUnverionedDocs = async () => {
+  const walk = async (dirPath: string): Promise<void> => {
+    const entries = await fsPromise.readdir(dirPath, {withFileTypes: true});
+    await Promise.all(
+      entries.map(async (entry) => {
+        const childPath = pathlib.join(dirPath, entry.name);
+        if (entry.isDirectory()) {
+          return walk(childPath);
+        } else {
+          transformFile(childPath);
+        }
+      })
+    );
+  };
+
+  console.log('Starting build-unversioned-docs.js script');
+  await walk(LATEST_VERSION_CONTENT);
+  console.log('Completed build-unversioned-docs.js script');
+};
+
+/**
+ * Transform the given versioned content file and copy it to the unversioned
+ * directory.
+ */
+function transformFile(path: string) {
+  let fileContents = fs.readFileSync(path, {encoding: 'utf8'});
+  const relativeChildPath = pathlib.relative(LATEST_VERSION_CONTENT, path);
+  const ext = pathlib.extname(relativeChildPath);
+  let unversionedLocation = pathlib.join(
+    UNVERSIONED_VERSION_LOCATION,
+    relativeChildPath
+  );
+
+  if (ext === '.md') {
+    const fileName = pathlib.basename(unversionedLocation, '.md');
+    const [frontMatterData, restOfFile] = getFrontMatterData(fileContents);
+    const existingPermalink = frontMatterData.findIndex((val) =>
+      val.includes('permalink:')
+    );
+    if (existingPermalink !== -1) {
+      throw new Error(
+        'Unhandled case: Handle this by transforming the permalink here.'
+      );
+    }
+    if (fileName === 'index') {
+      frontMatterData.push(
+        `permalink: docs/${relativeChildPath.slice(0, -3)}.html`
+      );
+    } else {
+      frontMatterData.push(
+        `permalink: docs/${relativeChildPath.slice(0, -3)}/index.html`
+      );
+    }
+
+    fileContents = writeFrontMatter(frontMatterData) + restOfFile;
+  } else if (ext === '.json') {
+    if (
+      pathlib.basename(unversionedLocation, '.json') === SITE_LATEST_VERSION
+    ) {
+      unversionedLocation = pathlib.join(
+        pathlib.dirname(unversionedLocation),
+        'unversioned.json'
+      );
+      fileContents = JSON.stringify({
+        collection: 'docs-unversioned',
+        latestVersion: SITE_LATEST_VERSION,
+      });
+    }
+  } else if (ext === '.html') {
+    if (pathlib.basename(unversionedLocation, '.html') === 'api') {
+      const [frontMatterData, _] = getFrontMatterData(fileContents);
+      const transformedFrontMatter = frontMatterData.map((line) => {
+        return line.replace(`/${SITE_LATEST_VERSION}/`, '/');
+      });
+      fileContents = writeFrontMatter(transformedFrontMatter);
+    } else {
+      throw new Error(`Unhandled html document: '${path}'`);
+    }
+  } else {
+    throw new Error(`Unhandled extension '${ext}' for '${path}'`);
+  }
+
+  fs.mkdirSync(pathlib.dirname(unversionedLocation), {recursive: true});
+  fs.writeFileSync(unversionedLocation, fileContents);
```

Comment:
Would it be too noisy to log something like `Writing ${unversionedLocation}...`?

Reply:
Added! Great suggestion!